EmptyStatement: false negative on top-level empty declaration in compact source file

Found during compact source coverage work for #20590 (PR #20725).

I have executed the cli and showed it below.
```
$ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
    "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="EmptyStatement"/>
  </module>
</module>
```
Compact source file (JEP 512):
```
$ cat Compact.java
;

void main() {
    ;

    boolean cond = true;
    for (; cond;);

    if (cond);

    while (cond) {
        cond = false;
        ;
    }
}
```

Equivalent ordinary source for comparison:
```
$ cat Ordinary.java
public class Ordinary {
    ;

    void main() {
        ;

        boolean cond = true;
        for (; cond;);

        if (cond);

        while (cond) {
            cond = false;
            ;
        }
    }
}
```
```
$ java -Duser.language=en -jar checkstyle-13.0.1-SNAPSHOT-all.jar -c config.xml Compact.java
$ java -Duser.language=en -jar checkstyle-13.0.1-SNAPSHOT-all.jar -c config.xml Ordinary.java
```
Starting audit...
[ERROR] C:\....\cmp-emptystatement\Compact.java:4:5: Empty statement. [EmptyStatement]
[ERROR] C:\....\cmp-emptystatement\Compact.java:7:18: Empty statement. [EmptyStatement]
[ERROR] C:\....\cmp-emptystatement\Compact.java:9:14: Empty statement. [EmptyStatement]
[ERROR] C:\....\cmp-emptystatement\Compact.java:13:9: Empty statement. [EmptyStatement]
Audit done.
Checkstyle ends with 4 errors.


Starting audit...
[ERROR] C:\...\cmp-emptystatement\Ordinary.java:5:9: Empty statement. [EmptyStatement]
[ERROR] C:\...\cmp-emptystatement\Ordinary.java:8:22: Empty statement. [EmptyStatement]
[ERROR] C:\...\cmp-emptystatement\Ordinary.java:10:18:Empty statement. [EmptyStatement]
[ERROR] C:\...\cmp-emptystatement\Ordinary.java:14:13:Empty statement. [EmptyStatement]
Audit done.
Checkstyle ends with 4 errors.

In both cases only the `;` inside the method body is reported. The
top-level `;` (compact) and the `;` between class members (ordinary)
produce no `EMPTY_STAT` token, so the check does not react to them.
The behavior of compact and ordinary source is consistent with each
other — this issue is about the compact case specifically.

**Describe what you expect in detail.**

A violation should be reported for the top-level `;` at line 1 of the
compact source file. This empty declaration outside of any method is
accepted by the compiler but serves no purpose, so EmptyStatement
should flag it.

The input file in PR #20725 marks this line with `// ok until <this issue>`.

--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EmptyStatementCheck.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EmptyStatementCheck.java
@@ -53,12 +53,15 @@ public class EmptyStatementCheck extends AbstractCheck {
 
     @Override
     public int[] getRequiredTokens() {
-        return new int[] {TokenTypes.EMPTY_STAT};
+        return new int[] {TokenTypes.EMPTY_STAT, TokenTypes.SEMI};
     }
 
     @Override
     public void visitToken(DetailAST ast) {
-        log(ast, MSG_KEY);
+        if (ast.getType() == TokenTypes.EMPTY_STAT
+                || ast.getParent().getType() == TokenTypes.COMPACT_COMPILATION_UNIT) {
+            log(ast, MSG_KEY);
+        }
     }
 
 }

--- a/src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/EmptyStatementCheckTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/EmptyStatementCheckTest.java
@@ -77,6 +77,7 @@ public class EmptyStatementCheckTest
     @Test
     public void testCompactSourceFile() throws Exception {
         final String[] expected = {
+            "9:1: " + getCheckMessage(MSG_KEY),
             "12:5: " + getCheckMessage(MSG_KEY),
             "15:18: " + getCheckMessage(MSG_KEY),
             "17:14: " + getCheckMessage(MSG_KEY),

--- a/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/emptystatement/compact/InputEmptyStatementCompactSourceFile.java
+++ b/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/emptystatement/compact/InputEmptyStatementCompactSourceFile.java
@@ -6,7 +6,7 @@ EmptyStatement
 
 // non-compiled with javac: Compilable with Java25
 
-; // ok until https://github.com/checkstyle/checkstyle/issues/20749
+; // violation 'Empty statement'
 
 void main() {
     ; // violation 'Empty statement'

